Can't build template with vulkan disabled

### Godot version

4.0.dev

### System information

Ubuntu 22.10, GTX1060

### Issue description

Trying to build **OpenGL only** template. Catched error:

```bash
scons: Reading SConscript files ...
Warning: fontconfig libraries not found. Disabling the system fonts support.
Warning: D-Bus development libraries not found. Disabling screensaver prevention.
Warning: Speech Dispatcher development libraries not found. Disabling Text-to-Speech support.
collect2 version 12.2.0
/usr/bin/ld -plugin /usr/lib/gcc/x86_64-linux-gnu/12/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-linux-gnu/12/lto-wrapper -plugin-opt=-fresolution=/tmp/cczJX5TF.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr -m elf_x86_64 --hash-style=gnu --as-needed -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie -z now -z relro /usr/lib/gcc/x86_64-linux-gnu/12/../../../x86_64-linux-gnu/Scrt1.o /usr/lib/gcc/x86_64-linux-gnu/12/../../../x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/12/crtbeginS.o -L/usr/lib/gcc/x86_64-linux-gnu/12 -L/usr/lib/gcc/x86_64-linux-gnu/12/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/12/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/12/../../.. --version -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-linux-gnu/12/crtendS.o /usr/lib/gcc/x86_64-linux-gnu/12/../../../x86_64-linux-gnu/crtn.o
Building for platform "linuxbsd", architecture "x86_64", target "template_debug".
Using build feature profile: fl.build
Checking for C header file mntent.h... (cached) yes
scons: done reading SConscript files.
scons: Building targets ...
[  1%] run(["core/object/gdvirtual.gen.inc"], ["core/object/make_virtuals.py"])
[  2%] Generating enabled modules header.
[  2%] Compiling platform/linuxbsd/joypad_linux.cpp ...
[  2%] Compiling platform/linuxbsd/freedesktop_screensaver.cpp ...
[  2%] Compiling platform/linuxbsd/os_linuxbsd.cpp ...
[  2%] Compiling platform/linuxbsd/crash_handler_linuxbsd.cpp ...
[  2%] Compiling platform/linuxbsd/godot_linuxbsd.cpp ...
In file included from platform/linuxbsd/x11/display_server_x11.h:58,
                 from platform/linuxbsd/os_linuxbsd.cpp:43:
./drivers/vulkan/rendering_device_vulkan.h:45:10: fatal error: vk_mem_alloc.h: No such file or directory
   45 | #include "vk_mem_alloc.h"
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
scons: *** [platform/linuxbsd/os_linuxbsd.linuxbsd.template_debug.x86_64.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:00:02.409]
```

### Steps to reproduce

1. make custom.build file with following:

```json
{
	"disabled_build_options": {
		"disable_3d": true,
		"disable_3d_physics": true,
		"openxr": false,
		"rendering_device": false,
		"vulkan": false
	},
	"disabled_classes": [

	],
	"type": "build_profile"
}
```
2. run following command on terminal
```
scons platform=linux target=template_debug tools=no build_feature_profile=custom.build -j4

```

### Minimal reproduction project

there is no need for mrp

--- a/SConstruct
+++ b/SConstruct
@@ -202,7 +202,7 @@ opts.Add("extra_suffix", "Custom extra suffix added to the base filename of all 
 opts.Add(BoolVariable("vsproj", "Generate a Visual Studio solution", False))
 opts.Add(BoolVariable("disable_3d", "Disable 3D nodes for a smaller executable", False))
 opts.Add(BoolVariable("disable_advanced_gui", "Disable advanced GUI nodes and behaviors", False))
-opts.Add("build_feature_profile", "Path to a file containing a feature build profile", "")
+opts.Add("build_profile", "Path to a file containing a feature build profile", "")
 opts.Add(BoolVariable("modules_enabled_by_default", "If no, disable all modules except ones explicitly enabled", True))
 opts.Add(BoolVariable("no_editor_splash", "Don't use the custom splash screen for the editor", True))
 opts.Add("system_certs_path", "Use this path as SSL certificates default for editor (for package maintainers)", "")
@@ -491,6 +491,25 @@ if selected_platform in platform_list:
     LINKFLAGS = env.get("LINKFLAGS", "")
     env["LINKFLAGS"] = ""
     env.Append(LINKFLAGS=str(LINKFLAGS).split())
+
+    # Feature build profile
+    disabled_classes = []
+    if env["build_profile"] != "":
+        print("Using feature build profile: " + env["build_profile"])
+        import json
+
+        try:
+            ft = json.load(open(env["build_profile"]))
+            if "disabled_classes" in ft:
+                disabled_classes = ft["disabled_classes"]
+            if "disabled_build_options" in ft:
+                dbo = ft["disabled_build_options"]
+                for c in dbo:
+                    env[c] = dbo[c]
+        except:
+            print("Error opening feature build profile: " + env["build_profile"])
+            Exit(255)
+    methods.write_disabled_classes(disabled_classes)
 
     # Platform specific flags.
     # These can sometimes override default options.
@@ -817,26 +836,6 @@ if selected_platform in platform_list:
         env["LIBSUFFIXES"] += [env["LIBSUFFIX"], env["SHLIBSUFFIX"]]
     env["LIBSUFFIX"] = suffix + env["LIBSUFFIX"]
     env["SHLIBSUFFIX"] = suffix + env["SHLIBSUFFIX"]
-
-    disabled_classes = []
-
-    if env["build_feature_profile"] != "":
-        print("Using build feature profile: " + env["build_feature_profile"])
-        import json
-
-        try:
-            ft = json.load(open(env["build_feature_profile"]))
-            if "disabled_classes" in ft:
-                disabled_classes = ft["disabled_classes"]
-            if "disabled_build_options" in ft:
-                dbo = ft["disabled_build_options"]
-                for c in dbo:
-                    env[c] = dbo[c]
-        except:
-            print("Error opening feature build profile: " + env["build_feature_profile"])
-            Exit(255)
-
-    methods.write_disabled_classes(disabled_classes)
 
     if env["disable_3d"]:
         if env.editor_build:

